src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -59,7 +60,24 @@ protected override void ExecuteCore()
             {
                 Directory.CreateDirectory(destinationDirectory);
             }
-            File.WriteAllBytes(ModifiedAppHostPath, array);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
The 'if' should not be necessary, although keeping it will prevent the (unnecessary) file copy for Windows.